Sources/NIO/ByteBuffer-conversions.swift

```diff
@@ -0,0 +1,33 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2017-2020 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+extension Array where Element == UInt8 {
+    
+    /// Creates a `[UInt8]` from the given buffer. The entire readable portion of the buffer will be read.
+    /// - parameter buffer: The buffer to read.
+    init(from buffer: ByteBuffer) {
+        self = Array(buffer.readableBytesView)
+    }
+    
+}
+
+extension String {
+    
+    /// Creates a `String` from a given `ByteBuffer`. The entire readable portion of the buffer will be read.
+    /// - parameter buffer: The buffer to read.
+    public init(from buffer: ByteBuffer) {
+        self = String(decoding: buffer.readableBytesView, as: Unicode.UTF8.self)
```

Comment:
what does this return when buffer is empty?

Reply:
empty string

Reply:
this should also probably be faster using `read/getString`

Reply:
not sure we need encodings tbh

Reply:
I actually expected this method to be faster, which is why I used it. If you're confident readString is faster I can switch :)

Reply:
very much, https://bugs.swift.org/browse/SR-12125

Reply:
@glbrntt I don't think you can specify the encoding in `readString`, unless I'm not looking in the right place...

Reply:
`readString` with encoding is there, but isn't useful to us because it's in the FoundationCompat target. We're going to punt for a while and address in a future PR